prow/kube/prowjob.go

```diff
@@ -95,6 +95,29 @@ type ProwJobSpec struct {
 	RunAfterSuccess []ProwJobSpec `json:"run_after_success,omitempty"`
 }
 
+
+const (
+	PathStrategyLegacy   = "legacy"
+	PathStrategySingle   = "single"
+	PathStrategyExplicit = "explicit"
+)
+
+// GCSConfiguration holds options for pushing logs and
+// artifacts to GCS from a job.
+type GCSConfiguration struct {
```

Comment:
Added this here instead of in `prow/gcsupload` like the other types of options as this is (eventually going to be) user-facing